src/workspace-data/useDataTableProvider.ts

```diff
@@ -33,98 +37,122 @@ export const useDataTableProvider = (
+    // setWdsTypes({ status: 'None' }); //TODO I don't think we need this since we initialize it to None
+    if (wdsUrl.status === 'Ready') {
       return Ajax(signal)
-        .WorkspaceData.getSchema(url, workspaceId)
+        .WorkspaceData.getSchema(wdsUrl.state, workspaceId)
         .then((typesResult) => {
           setWdsTypes({ status: 'Ready', state: typesResult });
         })
         .catch((error) => {
           setWdsTypes({ status: 'Error', state: null, error: 'Error loading WDS schema' });
           reportError('Error loading WDS schema', error);
         });
-    },
-    [signal, setWdsTypes]
-  );
+    }
+  }, [signal, setWdsTypes, wdsUrl, workspaceId]);
 
-  const loadWdsCapabilities = useCallback(
-    async (url) => {
+  const loadWdsCapabilities = useCallback(async () => {
+    if (wdsUrl.status === 'Ready') {
       try {
-        const capabilitiesResult = await Ajax(signal).WorkspaceData.getCapabilities(url);
+        const capabilitiesResult = await Ajax(signal).WorkspaceData.getCapabilities(wdsUrl.state);
         setWdsCapabilities({ status: 'Ready', state: capabilitiesResult });
       } catch (error) {
         setWdsCapabilities({ status: 'Error', state: null, error: 'Error loading WDS capabilities' });
         reportError('Error loading WDS capabilities', error);
       }
-    },
-    [signal, setWdsCapabilities]
-  );
+    }
+  }, [signal, setWdsCapabilities, wdsUrl]);
 
   const loadWdsData = useCallback(async () => {
-    // Try to load the proxy URL
-    if (!wdsApp || !['Ready', 'Error'].includes(wdsApp.status)) {
-      const foundApp = await loadWdsApp(workspaceId);
-      // TODO: figure out how not to make this redundant fetch, per
-      // https://github.com/DataBiosphere/terra-ui/pull/4202#discussion_r1319145491
-      if (foundApp) {
-        loadWdsTypes(foundApp.proxyUrls?.wds, workspaceId);
-        loadWdsCapabilities(foundApp.proxyUrls?.wds);
+    if (useCwds.status === 'Ready') {
+      // Try to load the proxy URL
+      if (!['Ready', 'Error'].includes(wdsApp.status)) {
+        await loadWdsApp();
       }
     }
-    // If we have the proxy URL try to load the WDS types
-    else if (wdsApp?.status === 'Ready') {
-      const proxyUrl = wdsApp.state.proxyUrls?.wds;
-      await loadWdsTypes(proxyUrl, workspaceId);
-      await loadWdsCapabilities(proxyUrl);
+  }, [wdsApp, loadWdsApp, useCwds]);
+
+  useEffect(() => {
+    if (useCwds.status === 'Ready' && wdsApp.status === 'None') {
+      loadWdsApp();
+    }
+    if (wdsUrl.status === 'Ready') {
+      loadWdsTypes();
+      loadWdsCapabilities();
+    }
+  }, [wdsUrl, loadWdsTypes, loadWdsCapabilities, useCwds, wdsApp, loadWdsApp]);
+
+  useEffect(() => {
+    const checkCWDS = async (): Promise<void> => {
+      try {
+        const response = await fetchWDS(cwdsURL)(`collections/v1/${workspaceId}`, _.merge(authOpts(), { signal }));
+        const data = await response.json();
+        if (Object.keys(data).length !== 0) {
```

Comment:
I'm a little worried about this simple `length` check. What if cWDS returned an error but the error was nicely-formatted json? Will this check still pass?

Reply:
Presumably if it returned an error the status code would also be an error code?  I can also make sure it's 200